services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -2423,7 +2423,7 @@ public void updateStore(String clusterName, String storeName, UpdateStoreQueryPa
               .orElseGet(currStore::isWriteComputationEnabled);
       setStore.replicationMetadataVersionID = replicationMetadataVersionID
           .map(addToUpdatedConfigList(updatedConfigsList, REPLICATION_METADATA_PROTOCOL_VERSION_ID))
-          .orElse(currStore.getRmdVersionID().orElse(-1));
+          .orElse(currStore.getRmdVersion());
```

Comment:
in this case the value is  0 and not -1 (because zero is the default value)

Reply:
`Store#getRmdVersionID` from all code paths was ending up in `ZkStore#getRmdVersionID` which had the logic:
```
@Override
  public Optional<Integer> getRmdVersionID() {
    final int rmdVersionID = this.storeProperties.replicationMetadataVersionID;
    return rmdVersionID == -1 ? Optional.empty() : Optional.of(rmdVersionID);
  }
```

Since we changed 
```

Reply:
I think what Enrico is indicating is that Jackson might default to 0 for an int property that isn't present in the payload. That may be true in general (IDK) but I verified that it is not the case in this particular implementation because in the `ZkStore` constructor we do:

```
this.storeProperties = Store.prefillAvroRecordWithDefaultValue(new StoreProperties());
```

And so the property of interest ends up getting filled with Avro's default (-1 in this case).

I added a test to assert this in the latest commit. Also modified another test to ensure that when the controller finds -1 in this store-level config, it will default to the controller-level config.

So in summary, I think the implementation in the original commit was already correct, but I've tweaked the unit tests to ensure it, and to ensure we catch regressions in those areas.

Thanks!